### ISSUE TITLE
Push and receive MQTT?

### ISSUE THREAD

**Issue by kalaws** [qix67/comfortzone_heatpump#5]
I'm struggling with how to properly configure all values to be published through MQTT. Would you care to share your mqtt configuration of the esp8266?

Also, any idea how sending updated values would work for MQTT?

Truly appreciate the time you're putting into this!

**Comment by qix67** [comment]
Not sure my code will help you because it depends on what MQTT client expect.
[mqtt_com.txt](https://github.com/qix67/comfortzone_heatpump/files/10089581/mqtt_com.txt)

My code has 4 main functions:
* mqtt_init() : to call during initialization
* mqtt_process(): to call from main_loop()
* mqtt_publish(): to send data to MQTT broker
* mqtt_register_custom_callback(): to register a callback associated to a task and a target. These callbacks will be used according to "task" and "target" to process message received from MQTT broker

my MQTT message are JSON encoded with at least 2 variables named "task" and "target".

manage_ctrl() gives you an example on how I send data.

**Comment by kalaws** [comment]
Thanks. This is far above my skills, so I hope you’ll bear with me. Still grappling even with how to setup calls asking heatpump for data. You supplied an example file, how would I go about using that? I.e, through where do I initiate those functions?

**Comment by qix67** [comment]
To avoid busying RS485 bus uselessly, I do not ask anything to heatpump because the control panel already does it and the library catch reply on the fly.

In the example file, if you want to change fan speed to fast speed (for example), just do:
```
heatpump.set_fan_speed(3);
```

If fan speed is not modified, after heatpump.begin(), you may have to call `heatpump.guess_heatpump_addr()`. I don't know if all heatpump share the same address on bus. My default address is hardcoded but it may not be the correct address for you.

**Comment by kalaws** [comment]
So if I understand correctly, all relevant status will by default be parsed by the library and visibile in the esphome console, for instance? I.e., I don't have to make any configuration of the library for the read function to work? Stuff should be arriving mid next week, perhaps things will seem clearer to me once it is hooked up. I managed to get a read function to work with my hold heat pump, so I'll be damned if I can't get this to work...

**Comment by qix67** [comment]
Yes, I have this. My esp8266 provides a small webpage reporting all parsed values, most of them are not sent to domoticz.

**Comment by GitBaer** [comment]
Hi Qix! Could you please share the Webpage-Code to test? Thank you very much :)

**Comment by qix67** [comment]
Here it is. I am not sure it can be useful but here it is. It contains 3 things: status webpage, mqtt report and mqtt control
[input_cmd.txt](https://github.com/qix67/comfortzone_heatpump/files/10101701/input_cmd.txt)

**Comment by kalaws** [comment]
A related question. I see you have an `include` referencing sensors. I’m kinda struggling with how to expose sensors to esphome, but perhaps that is not what that include does?

**Comment by qix67** [comment]
As I don't use esphome, I can't help you.

Include file contains IP and sensors lists of all my µC, nothing useful for you there.
